packages/jsts/src/rules/S1607/rule.ts

```diff
@@ -234,14 +234,21 @@ export const rule: Rule.RuleModule = {
     }
 
     /**
-     * Checks if the node denoting a test has an explanation comment on the line above.
+     * Checks if the node denoting a test has an adjacent explanation comment.
      */
-    function hasExplanationComment(node: estree.Node): boolean {
-      const comments = context.sourceCode.getCommentsBefore(node);
-      return comments.some(
-        comment =>
-          comment.loc!.end.line === node.loc!.start.line - 1 && comment.value.trim().length > 0,
-      );
+    function hasExplanationComment(node: estree.CallExpression) {
+      function isAdjacent(comment: estree.Comment, node: estree.Node) {
+        const commentLine = comment.loc!.end.line;
+        const nodeLine = node.loc!.start.line;
+        return commentLine === nodeLine - 1 || commentLine === nodeLine + 1;
```

Comment:
If we consider [my comment here](https://github.com/SonarSource/SonarJS/pull/4804#issuecomment-2334112099),

This could be:

```suggestion
        return Math.abs(commentLine - nodeLine) <= 1;
```

Reply:
I'm not sure how it could be tested in the comment-based framework though.

Reply:
Yeah, I was going to, but I just don't know how to test that.
Any suggestions?

Reply:
Another option is to use relative line increments (`@+line_increment`) or decrements (`@-line_decrement`):

```javascript
// Noncompliant@+1
some.faulty.code();

another.faulty.code();
// another comment
// Noncompliant@-2
```

from DEV.md

Reply:
Thank you so much for remembering it!